app/js/views/themes-new.jsx

```diff
@@ -12,120 +12,118 @@ export default class ThemesNew extends BaseView {
 
     this.updateServices = this.updateServices.bind(this);
-    this.handleServiceWithGetChange = this.handleServiceWithGetChange
-      .bind(this);
-    this.handleGetterChange = this.handleGetterChange.bind(this);
-    this.handleServiceWithSetChange = this.handleServiceWithSetChange
-      .bind(this);
-    this.handleSetterChange = this.handleSetterChange.bind(this);
+    this.onGetterSelected = this.onGetterSelected.bind(this);
+    this.onGetterValueSelected = this.onGetterValueSelected.bind(this);
+    this.onSetterSelected = this.onSetterSelected.bind(this);
+    this.onSetterValueSelected = this.onSetterValueSelected.bind(this);
     this.handleActionButton = this.handleActionButton.bind(this);
   }
 
   componentDidMount() {
     Promise.all([
-        this.foxbox.recipes.getServicesWithGetters(),
-        this.foxbox.recipes.getServicesWithSetters()
-      ])
-      .then(services => {
-        this.updateServices(services);
-      })
-      .catch(console.error.bind(console));
+      this.foxbox.recipes.getGetters(),
+      this.foxbox.recipes.getSetters()
+    ])
+    .then(services => this.updateServices(services))
+    .catch(console.error.bind(console));
   }
 
-  updateServices([servicesWithGet, servicesWithSet] = [[], []]) {
-    this.setState({ servicesWithGet, servicesWithSet });
+  updateServices([getters, setters] = [[], []]) {
+    this.setState({ getters, setters });
   }
 
-  handleServiceWithGetChange(evt) {
-    const serviceWithGet = evt.target.value;
-    if (serviceWithGet !== '') {
-      this.setState({ serviceWithGet });
+  onGetterSelected(evt) {
+    const selectedGetterIndex = evt.target.value;
+    if (selectedGetterIndex) {
+      this.setState({ selectedGetterIndex: +selectedGetterIndex });
     } else {
       this.setState({
-        serviceWithGet: null,
-        getter: null,
-        serviceWithSet: null,
-        setter: null
+        selectedGetterIndex: -1,
+        selectedGetterValueIndex: -1,
+
+        selectedSetterIndex: -1,
+        selectedSetterValueIndex: -1
       });
     }
   }
 
-  handleGetterChange(evt) {
-    const getter = evt.target.value;
-    if (getter !== '') {
-      this.setState({ getter });
+  onGetterValueSelected(evt) {
+    const selectedGetterValueIndex = evt.target.value;
+    if (selectedGetterValueIndex) {
+      this.setState({ selectedGetterValueIndex: +selectedGetterValueIndex });
     } else {
       this.setState({
-        getter: null,
-        serviceWithSet: null,
-        setter: null
+        selectedGetterValueIndex: -1,
+
+        selectedSetterIndex: -1,
+        selectedSetterValueIndex: -1
       });
     }
   }
 
-  handleServiceWithSetChange(evt) {
-    const serviceWithSet = evt.target.value;
-    if (serviceWithSet !== '') {
-      this.setState({ serviceWithSet });
+  onSetterSelected(evt) {
+    const selectedSetterIndex = evt.target.value;
+    if (selectedSetterIndex) {
+      this.setState({ selectedSetterIndex: +selectedSetterIndex });
```

Comment:
Is the `+` in front of `selectedSetterIndex` to cast it to a Number?
If so, I prefer this to be made in the `const` assignation. Maybe also use something more noticeable (like `Number(selectedSetterIndex)` or `parseInt(selectedSetterIndex)`).